src/uu/echo/src/echo.rs

```diff
@@ -39,77 +47,117 @@ impl Base {
+        Some(&&ue) => match char::from(ue).to_digit(radix_u_three_two) {
+            // A u8 interpreted as a hexadecimal or octal digit is never more than 16
+            Some(ut) => u8::try_from(ut).unwrap(),
+            None => {
+                return None;
+            }
+        },
+        None => {
+            return None;
+        }
+    };
 
     // We can safely ignore the None case because we just peeked it.
     let _ = input.next();
 
     for _ in 1..base.max_digits() {
-        match input.peek().and_then(|c| c.to_digit(base as u32)) {
-            Some(n) => ret = ret.wrapping_mul(base as u8).wrapping_add(n as u8),
-            None => break,
+        match input
+            .peek()
+            .and_then(|&&ue| char::from(ue).to_digit(radix_u_three_two))
+        {
+            Some(ut) => {
+                // A u8 interpreted as a hexadecimal or octal digit is never more than 16
+                let ue = u8::try_from(ut).unwrap();
+
+                sum = sum.wrapping_mul(radix).wrapping_add(ue)
+            }
+            None => {
+                break;
+            }
         }
+
         // We can safely ignore the None case because we just peeked it.
         let _ = input.next();
     }
 
-    Some(ret.into())
+    Some(sum)
 }
 
-fn print_escaped(input: &str, mut output: impl Write) -> io::Result<ControlFlow<()>> {
-    let mut iter = input.chars().peekable();
-    while let Some(c) = iter.next() {
-        if c != '\\' {
-            write!(output, "{c}")?;
+fn print_escaped(input: &[u8], output: &mut StdoutLock) -> io::Result<ControlFlow<()>> {
+    let mut iter = input.iter().peekable();
+
+    while let Some(&current_byte) = iter.next() {
+        if current_byte != b'\\' {
+            output.write_all(&[current_byte])?;
+
             continue;
         }
 
         // This is for the \NNN syntax for octal sequences.
         // Note that '0' is intentionally omitted because that
         // would be the \0NNN syntax.
-        if let Some('1'..='8') = iter.peek() {
+        if let Some(b'1'..=b'8') = iter.peek() {
             if let Some(parsed) = parse_code(&mut iter, Base::Oct) {
-                write!(output, "{parsed}")?;
+                output.write_all(&[parsed])?;
+
                 continue;
             }
         }
 
         if let Some(next) = iter.next() {
-            let unescaped = match next {
-                '\\' => '\\',
-                'a' => '\x07',
-                'b' => '\x08',
-                'c' => return Ok(ControlFlow::Break(())),
-                'e' => '\x1b',
-                'f' => '\x0c',
-                'n' => '\n',
-                'r' => '\r',
-                't' => '\t',
-                'v' => '\x0b',
-                'x' => {
-                    if let Some(c) = parse_code(&mut iter, Base::Hex) {
-                        c
+            // For extending lifetime
+            let sl: [u8; 1_usize];
+            let sli: [u8; 2_usize];
```

Comment:
Why are those variables needed? Plus the variable names are not that great ;-)

Reply:
Will fix the names. It's an MSRV issue. Until Rust 1.79, temporary lifetime extension didn't work in if and match:

https://github.com/rust-lang/rust/pull/121346
https://releases.rs/docs/1.79.0/

Reply:
Ah, ok. Can you please add a todo to remove those vars in the future when we have an MSRV of 1.79?